wgengine/magicsock/derp.go

```diff
@@ -178,6 +178,9 @@ func (c *Conn) maybeSetNearestDERP(report *netcheck.Report) (preferredDERP int)
 		// one.
 		preferredDERP = c.pickDERPFallback()
 	}
+	if preferredDERP != c.myDerp {
+		c.logf("magicsock: home DERP changing from derp-%d [%dms] to derp-%d [%dms]", c.myDerp, report.RegionLatency[c.myDerp].Milliseconds(), preferredDERP, report.RegionLatency[preferredDERP].Milliseconds())
```

Comment:
go technically doesn't have line length limits, but this is long enough, especially compared to surrounding code, that' I'd probably go ahead and split after the format string.

I think we should probably also move this inside of setNearestDERP, mainly because we seem to always acquire the `c.mu` lock any time we touch `c.myDERP` (see line 163 above, for example).  setNearestDERP is already acquiring that lock, and it's already doing the comparison between the old and new value (see line 214 below).  So I think we could just add this log statement after line 217 (after it returns early if the preferred DERP matches the existing one), and you'll get the same effect as what you have here.

Reply:
oh, I was just thinking something as simple as:

``` go
c.logf("magicsock: home DERP changing from derp-%d [%dms] to derp-%d [%dms]", 
    c.myDerp, report.RegionLatency[c.myDerp].Milliseconds(), preferredDERP, report.RegionLatency[preferredDERP].Milliseconds())
```

😄 

As for the lock, I think we could pull the lock above out of the if statement...

```go
	c.mu.Lock()
	myDerp := c.myDerp
	c.mu.Unlock()

	if !connectedToControl {
		if myDerp != 0 {
			metricDERPHomeNoChangeNoControl.Add(1)
			return myDerp
		}

		// Intentionally fall through; we don't have a current DERP, so
		// as mentioned above selecting one even if not connected is
		// strictly better than doing nothing.
	}
```

So then the `myDERP` var is in scope for your logging and you don't have to acquire the lock again.